src/Conversion/ONNXToTorch/NN/PaddingOp.cpp

```diff
@@ -110,79 +107,144 @@ class ONNXConstantPadNdOpToTorchLowering : public ConversionPattern {
 
     ONNXPadOp op1 = llvm::dyn_cast<ONNXPadOp>(op);
     ONNXPadOpAdaptor adapter = ONNXPadOpAdaptor(op1);
-    mlir::MLIRContext *context =  op1.getContext();
+    mlir::MLIRContext *context = op1.getContext();
     Location loc = op1.getLoc();
 
-    Value data = op1.data();				// ONNX operands
-    auto pads = op1.pads();				// ONNX operands
-    Value const_value = op1.constant_value();		// ONNX operands
-
-    llvm::outs() << "pads type" << "\n" << pads.getType() << "\n" << "\n";
-    llvm::outs() << "pads data " << "\n" << pads << "\n" << "\n";
-
-    auto one = 1;
-    auto two = 2;
-    auto zero = 0;
-    auto three = 3;
+    Value data = op1.data();                  // ONNX operands
+    auto pads = op1.pads();                   // ONNX operands
+    Value const_value = op1.constant_value(); // ONNX operands
+
+    llvm::outs() << "pads type"
+                 << "\n"
+                 << pads.getType() << "\n"
+                 << "\n";
+    llvm::outs() << "pads data "
+                 << "\n"
+                 << pads << "\n"
+                 << "\n";
+    llvm::outs() << "constant_value "
+                 << "\n"
+                 << const_value << "\n"
+                 << "\n";
+
+    DenseElementsAttr denseAttr = getDenseElementAttributeFromONNXValue(pads);
+
+    // Reading the ONNX side pads values and store in the array.
+    std::vector<APInt> intValues;
+    for (auto n : denseAttr.getValues<APInt>())
+      intValues.push_back(n);
+
+    // Rearrange the pad values.
+    // ONNX : b1, e1, b2, e2, b3, e3, b4, e4
+    // TORCH : b1, b2, b3, b4, e1, e2, e3, e4
```

Comment:
```suggestion
```

The torch-mlir ordering is even more different than this. I've adjusted the code to get my example to work, but please sanity-check.

Reply:
The new changes look good to me. I've run it on the next const_pad_nd test (with different pads values in each run).